muxinc/mux-stats-sdk-avplayer
# bugfix - programChange and dispatch viewend when player is destroyed

Two changes here:

1. fix programChange
1. when calling `destroyPlayer`, let's dispatch `viewend` - it looks like an oversight that we were not doing this

----

When calling programChange, we need to call monitorAVPlayerItem on the playerBinding instance so that the binding will call videoChangedForPlayer on the playDispatchDelegate, otherwise the program change will not be properly tracked

QA:

1. Try using `programChangeForPlayer` on current master (v1.2.1) - it does not work properly. It ends up sending the first view twice 😑 
1. Try using `programChangeForPlayer` on this branch - it should work now

I'm especially keen on getting @nidhik 's review of this implementation because this touches some stuff she worked on in #35 and #36 

TODO for @dylanjha:

- [x] write a test for this
